cfme/tests/cloud_infra_common/test_cockpit_server.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+import pytest
+
+from cfme.cloud.instance import Instance
+from cfme.cloud.provider import CloudProvider
+from cfme.cloud.provider.ec2 import EC2Provider
+from cfme.cloud.provider.azure import AzureProvider
+from cfme.cloud.provider.gce import GCEProvider
+from cfme.common.provider import CloudInfraProvider
+from cfme.infrastructure.virtual_machines import Vm
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.generators import random_vm_name
+
+
+@pytest.fixture(scope="function")
+def new_vm(provider, request):
+    if provider.one_of(CloudProvider):
+        vm = Instance.factory(random_vm_name(context='cockpit'), provider)
+    else:
+        vm = Vm.factory(random_vm_name(context='cockpit'), provider)
+    if not provider.mgmt.does_vm_exist(vm.name):
+        vm.create_on_provider(find_in_cfme=True, allow_skip="default")
+        request.addfinalizer(vm.delete_from_provider)
+    return vm
+
+
+@pytest.mark.uncollectif(
+    lambda appliance, provider: appliance.version < "5.9" or provider.one_of(GCEProvider))
+@pytest.mark.provider([CloudInfraProvider])
+@pytest.mark.parametrize('enable', [False, True], ids=['disabled', 'enabled'])
+def test_cockpit_server_role(appliance, provider, setup_provider, new_vm, enable):
+    """ The test checks the cockpit "Web Console" button enable and disabled working. """
+
+    if enable:
```

Comment:
**REQUIRED:** We need also to set it to disabled if indeed enable is ```False```. We must ensure initial conditions.

Reply:
@psav Updated chnages